idx/idx.go

```diff
@@ -29,6 +34,43 @@ type Archive struct {
 	LastSave uint32 // last time the metricDefinition was saved to a backend store (cassandra)
 }
 
+type MetricID struct {
+	org   int
+	part1 uint64
+	part2 uint64
```

Comment:
probably makes more sense to model this as a `[16]byte`

Reply:
Does that not make the conversions more complicated? I'm researching what's the best way to go from a string to a `[16]byte`, and it seems like this is requiring to still parse the string into two `uint64` and then use something like `binary.PutUvarint(buf, x)` to convert them into `[8]byte` and then concatenate the two with some bitshift & or. 
I agree it would be "nicer" to have `[16]byte` instead of the two `uint64`, but the conversion just seems more complicate for no real benefit. Please consider that this conversion might happen very often once we use the `MetricID` type to identify metrics in the main index, then it will literally be executed on every data point that arrives, so we might want to avoid any unnecessary steps in that conversion.

Reply:
check this out https://gist.github.com/Dieterbe/e95a9ffedcf7da9e2cefd96bab0ab726 -- perf looks comparable

Reply:
cool, thx, will use that, that's clearly better.

Reply:
https://github.com/grafana/metrictank/pull/729/commits/fd4ca4ac23762d76cb3a09f2973ed3f4facb6797